datasets/us/ofac/enforcement_actions/crawler.py

```diff
@@ -0,0 +1,291 @@
+    html_to_text_hash,
+)
+
+NAME_XPATH = "//span[@class='treas-page-title']/text()"
+CONTENT_XPATH = "//div[@id='block-ofac-content']//div[@class='content']"
+DATE_XPATH = "//div[@id='block-ofac-content']//div[@class='field__item']/text()"
+# Traveling back to summer of 2016 (the first full pdf enforcement was issued on July 5, 2016)
+MAX_AGE_DAYS = (date.today() - date(2016, 7, 8)).days
+
+Schema = Literal["Person", "Company", "LegalEntity"]
+Status = Literal[
+    "Filed",
+    "Dismissed",
+    "Settled",
+    "Final judgement",
+    "Other",
+]
+
+# Settlement Agreement
+# Penalty Notice to an Individual
+# Finding of Violation
+# Sanctions Compliance Guidance
+# Issuance of Sanctions Regulations
+# Civil Monetary Penalty
+# Release of OFAC Enforcement Information
+# Release​ of Civil Penalties Information
+# Recent OFAC Actions
+
+MODEL_VERSION = 1
+MIN_MODEL_VERSION = 1
+
+something_changed = False
+
+schema_field = Field(
+    description="Use LegalEntity if it isn't clear whether the entity is a person or a company."
+)
+address_field = Field(
+    default=[],
+    description=("The addresses or even just the districts/states of the defendant."),
+)
+status_field = Field(
+    description=(
+        "The status of the enforcement action notice."
+        " If `Other`, add the text used as the status in the source to `notes`."
+    )
+)
+notes_field = Field(default=None, description=("Only used if `status` is `Other`."))
+
+
+class RelatedCompany(BaseModel):
+    name: str
+    relationship: str
+
+
+class Defendant(BaseModel):
+    entity_schema: Schema = schema_field
+    name: str
+    aliases: List[str] = []
+    address: str | List[str] | None = address_field
+    country: str | List[str] | None = []
+    status: Status = status_field
+    notes: Optional[str] = notes_field
+    related_companies: List[RelatedCompany] = []
+    pdf_url: Optional[str] = None
+
+
+class Defendants(BaseModel):
+    defendants: List[Defendant]
+
+
+PROMPT = f"""
+Extract the defendants or entities subject to OFAC enforcement actions in the attached article.
+NEVER include relief defendants.
+NEVER infer, assume, or generate values that are not directly stated in the source text.
+
+Trading/D.B.A. names which follow a person name but look like company can just be
+aliases of the person. If the name is a person name, use `Person` as the entity_schema.
+
+Specific fields:
+
+- name: The null name of the entity precisely as expressed in the text. If an acronym follows the name in parentheses, include it as an alias and not as part of the name.
+- entity_schema: {schema_field.description}
+- address: {address_field.description}
+- country: Any countries the entity is indicated to reside, operate, or have been born or registered in. Leave empty if not explicitly stated.
+- status: {status_field.description}
+- notes: {notes_field.description}
+- related_companies: If the defendant is a person and a related company is mentioned in the source text, add it here.
```

Comment:
let's try and clarify this so that we only get related entities where the entity has an ownership or controlling relationship with the related entity. e.g. on https://ofac.treasury.gov/recent-actions/20190425 we got rosneft as related which is cool, but I'm not sure quite whether we want to represent basic business dealings like this. With `transacted or otherwise dealt in new debt` it's also not clear if the business relationship was with rosneft or just underwriting the debt for another provider.

See also https://opensanctions-corp.slack.com/archives/C07C5G9KZED/p1755704163127739 but I think if we can make it fairly sane we can keep it in.